### PR TITLE
Allow type parameter names with single letter, for example Foo<K, V>

### LINKED ISSUE #937: CA1715 (IdentifiersShouldHaveCorrectPrefix) should not fire for single letter type parameter names
#### Repro steps

Create a method with type param names with single letter, example `Foo<K, V>`

#### Expected behavior
No CA1715 fired.
I think it is extremely common practice to have type parameter names with a single letter. We should avoid firing CA1715 for such cases.

#### Actual behavior
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1715	Prefix generic type parameter name V with 'T'.	DiagnosticsTestUtilities	C:\roslyn-analyzers\src\Test\Utilities\MarkupTestFile.cs	173	Active
Warning	CA1715	Prefix generic type parameter name K with 'T'.	DiagnosticsTestUtilities	C:\roslyn-analyzers\src\Test\Utilities\MarkupTestFile.cs	173	Active
```

### PR DESCRIPTION
**Pull request by bkoelman**
Fixes #937.

### CODE REVIEW

File: src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/IdentifiersShouldHaveCorrectPrefixTests.cs [hunk 1]
```diff
@@ -145,11 +145,7 @@ public class Class6<TTypeParameter>
 {
 }
 ",
-                GetCA1715CSharpResultAt(4, 25, CA1715TypeParameterMessage, "V"),
                 GetCA1715CSharpResultAt(8, 32, CA1715TypeParameterMessage, "\u672C\u8A9E"),
-                GetCA1715CSharpResultAt(12, 31, CA1715TypeParameterMessage, "V"),
-                GetCA1715CSharpResultAt(14, 21, CA1715TypeParameterMessage, "V"),
-                GetCA1715CSharpResultAt(18, 24, CA1715TypeParameterMessage, "V"),
```
Comment (mavasani) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/IdentifiersShouldHaveCorrectPrefixTests.cs]: We should change existing tests to still fire by renaming its type params to be >1 length and add a new test to cover type params with length 1.

**Comment by mavasani** [comment]
LGTM modulo test changes

File: src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/IdentifiersShouldHaveCorrectPrefixTests.cs (hunk 1 above)
Comment (bkoelman) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/IdentifiersShouldHaveCorrectPrefixTests.cs]: Good point, I'll work on that.

Comment (bkoelman) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/IdentifiersShouldHaveCorrectPrefixTests.cs]: Tests updated, can you take a look?

**Comment by mavasani** [comment]
LGTM

**Pull request merged**